src/main/java/org/reactome/addlinks/fileprocessors/ensembl/EnsemblBioMartFileProcessor.java

```diff
@@ -0,0 +1,167 @@
+                                if (properArraySizeAndNecessaryColumnsContainData(tabSplit, 2,transcriptIdentifierIndex, proteinIdentifierIndex)) {
+                                    proteinToTranscripts = mapIdentifiers(proteinToTranscripts, tabSplit.get(proteinIdentifierIndex), tabSplit.get(transcriptIdentifierIndex));
+                                    // Add each mapping generated to the 'super' mapping
+                                    mappings.put(biomartFileSpecies + EnsemblBioMartUtil.PROTEIN_TO_TRANSCRIPTS_SUFFIX, proteinToTranscripts);
+                                }
+                                // 'transcriptToMicroarrays' mapping requires the 2nd (transcript) and 4th (microarray) values in the line.
+                                if (properArraySizeAndNecessaryColumnsContainData(tabSplit, 3,transcriptIdentifierIndex, identifierIndex)) {
+                                    transcriptToMicroarrays = mapIdentifiers(transcriptToMicroarrays, tabSplit.get(transcriptIdentifierIndex), tabSplit.get(identifierIndex));
+                                    // Add each mapping generated to the 'super' mapping
+                                    mappings.put(biomartFileSpecies + EnsemblBioMartUtil.TRANSCRIPT_TO_MICROARRAYS_SUFFIX, transcriptToMicroarrays);
+                                }
+                            }
+                        }
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+            });
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return mappings;
+    }
+
+    /**
+     * Maps one type of identifier to a List of another type of identifier
+     * @param identifierToIdentifierMapping Map<String, List<String>> mapping structure that holds all identifier mappings
+     * @param identifier1 String -- Identifier that will be the key to the List of the other type of identifiers
+     * @param identifier2 String -- Identifier that will be added to the List keyed to the other identifier
+     * @return identifierToIdentifierMapping that has been updated with new values.
+     */
+    private Map<String, List<String>> mapIdentifiers(Map<String, List<String>> identifierToIdentifierMapping, String identifier1, String identifier2) {
+        identifierToIdentifierMapping.computeIfAbsent(identifier1, k -> new ArrayList<>()).add(identifier2);
+        return identifierToIdentifierMapping;
+    }
+    //            rgpIdentifiersToRGPs.computeIfAbsent(identifier, k -> new ArrayList<>()).add(rgpInst);
+    // Checks that data exists at the supplied column indices.
+    private boolean properArraySizeAndNecessaryColumnsContainData(List<String> tabSplit, int necessaryArrayLength, int necessaryColumnIndex1, int necessaryColumnIndex2) {
+         return tabSplit.size() > necessaryArrayLength && !tabSplit.get(necessaryColumnIndex1).isEmpty() && !tabSplit.get(necessaryColumnIndex2).isEmpty();
+    }
```

Comment:
The second argument of `necessaryArrayLength` looks like it should be computable based on the necessary column indices passed to the method

```
// Checks that data exists at the supplied column indices.
private boolean properArraySizeAndNecessaryColumnsContainData(String bioMartLine,
    List<Integer> necessaryColumnIndices) {

    int maxRequiredArrayIndex = Collections.max(necessaryColumnIndices);
    
    return getColumnsInLine(bioMartLine).size() > maxRequiredArrayIndex &&
           allNecessaryColumnsContainData(bioMartLine, necessaryColumnIndices);
}

private List<String> getColumnsInLine(String bioMartFileLine) {
    return Arrays.asList(bioMartFileLine.split("\t"));
}

private boolean allNecessaryColumnsContainData(String bioMartLine, List<Integer> necessaryColumnIndices) {
    for (int necessaryColumnIndex : necessaryColumnIndices) {
        if (getColumnsInLine(bioMartLine).get(necessaryColumnIndex).isEmpty()) {
            return false;
        }
    }
    return true;
}
```

Reply:
Not exactly how you suggested but I compute the necessary array size within the function now instead of being passed hard-coded array length values